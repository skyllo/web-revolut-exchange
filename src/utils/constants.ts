import { Pocket } from '../types';

export const POCKET_TO_SYMBOL: Record<Pocket, string> = {
  EUR: '€',
  GBP: '£',
  USD: '$',
};

export const FETCH_RATE_INTERVAL_MS = 10000;
