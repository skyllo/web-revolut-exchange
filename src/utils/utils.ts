import { useEffect, useRef } from 'react';

export function toFixedDecimals(
  num: number,
  round: boolean = true,
  numDecimals: number = 2
): number {
  if (!round) {
    const numText = num.toString();
    if (numText.indexOf('.') > 0) {
      return Number(numText.slice(0, numText.indexOf('.') + numDecimals + 1));
    }
  }
  return Number(num.toFixed(numDecimals));
}

export function usePrevious(value: any) {
  const ref = useRef();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
}

export function Noop() {}
