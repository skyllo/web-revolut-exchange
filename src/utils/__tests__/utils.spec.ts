import { toFixedDecimals } from '../utils';

describe('toFixedDecimals()', () => {
  it('should limit to 2 decimal places', () => {
    const result = toFixedDecimals(1000.506, false, 2);
    expect(result).toMatchSnapshot();
  });

  it('should limit to 2 decimal places and round up', () => {
    const result = toFixedDecimals(1000.506, true, 2);
    expect(result).toMatchSnapshot();
  });

  it('should no contain any decimal places', () => {
    const result = toFixedDecimals(1000.506, false, 0);
    expect(result).toMatchSnapshot();
  });

  it('should no contain any decimal places and round up', () => {
    const result = toFixedDecimals(1000.506, true, 0);
    expect(result).toMatchSnapshot();
  });
});
