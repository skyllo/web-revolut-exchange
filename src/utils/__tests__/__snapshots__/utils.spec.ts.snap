// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toFixedDecimals() should limit to 2 decimal places 1`] = `1000.5`;

exports[`toFixedDecimals() should limit to 2 decimal places and round up 1`] = `1000.51`;

exports[`toFixedDecimals() should no contain any decimal places 1`] = `1000`;

exports[`toFixedDecimals() should no contain any decimal places and round up 1`] = `1001`;
